cpp/src/slice2java/Gen.h

```diff
@@ -8,89 +8,129 @@
+            std::int32_t tag,
+            const std::string& param,
+            bool marshal,
+            int& iter,
+            const std::string& customStream = "",
+            const MetadataList& metadata = MetadataList(),
+            const std::string& patchParams = "");
 
-        std::string getResultType(const OperationPtr&, const std::string&, bool, bool);
+        //
+        // Generate code to marshal or unmarshal a dictionary type.
+        //
+        static void writeDictionaryMarshalUnmarshalCode(
+            IceInternal::Output& out,
+            const std::string& package,
+            const DictionaryPtr& dict,
+            const std::string& param,
+            bool marshal,
+            int& iter,
+            bool useHelper,
+            const std::string& customStream = "",
+            const MetadataList& metadata = MetadataList());
 
-        void writeResultTypeMarshalUnmarshalCode(
+        //
+        // Generate code to marshal or unmarshal a sequence type.
+        //
+        static void writeSequenceMarshalUnmarshalCode(
+            IceInternal::Output& out,
+            const std::string& package,
+            const SequencePtr& seq,
+            const std::string& param,
+            bool marshal,
+            int& iter,
+            bool useHelper,
+            const std::string& customStream = "",
+            const MetadataList& metadata = MetadataList());
+
+        static void writeResultTypeMarshalUnmarshalCode(
             IceInternal::Output& out,
             const OperationPtr& op,
             const std::string& package,
             const std::string& streamName,
             const std::string& paramPrefix,
             bool isMarshalling);
 
-        void writeResultType(
+        static void writeResultType(
             IceInternal::Output&,
             const OperationPtr&,
             const std::string&,
             const std::optional<DocComment>&);
-        void writeMarshaledResultType(
+        static void writeMarshaledResultType(
             IceInternal::Output&,
             const OperationPtr&,
             const std::string&,
             const std::optional<DocComment>&);
 
-        void allocatePatcher(IceInternal::Output&, const TypePtr&, const std::string&, const std::string&);
-        std::string getPatcher(const TypePtr&, const std::string&, const std::string&);
-
-        /// Returns a vector of this operation's parameters with each of them formatted as 'paramType paramName'.
-        /// If 'internal' is true, the names will be prefixed with "iceP_".
-        std::vector<std::string>
-        getParamsProxy(const OperationPtr& op, const std::string& package, bool optionalMapping, bool internal = false);
-
-        /// Returns a vector of this operation's parameter's names in order.
-        /// If 'internal' is true, the names will be prefixed with "iceP_".
-        std::vector<std::string> getInArgs(const OperationPtr& op, bool internal = false);
```

Comment:
These 3 functions were helper functions, not code-gen.
They were moved to `JavaUtil`.